### PR TITLE
Custom events created

### PR DESCRIPTION
**Pull request by Abuelodelanada**
In this PR we created 
- `LokiDepartedEvent`: Fired when the relation is removed.
- `LokiPushApiEndpointJoined`: Fired when the relation is fully established

### CODE REVIEW

**Comment by mmanciop** [comment]
@Abuelodelanada I am wondering why we have the Departed event and not also its corresponding Joined event. What is the rationale?

**Review by mmanciop — APPROVED** [review]

File: lib/charms/loki_k8s/v0/loki_push_api.py [hunk 1]
```diff
@@ -457,6 +482,25 @@ def __init__(self, message="Alert rule is not well formed"):
         super().__init__(self.message)
 
 
+class LokiPushApiEndpointDeparted(EventBase):
+    """Event emitted when Loki departed."""
+
+    pass
```
Comment (sed-i) [inline lib/charms/loki_k8s/v0/loki_push_api.py:488]: `pass` is redundant when you already have a docstring.

File: lib/charms/loki_k8s/v0/loki_push_api.py [hunk 2]
```diff
@@ -457,6 +482,25 @@ def __init__(self, message="Alert rule is not well formed"):
...
+
+
+class LokiPushApiEndpointJoined(EventBase):
+    """Event emitted when Loki joined."""
+
+    pass
```
Comment (sed-i) [inline lib/charms/loki_k8s/v0/loki_push_api.py:494]: `pass` is redundant when you already have a docstring.

File: lib/charms/loki_k8s/v0/loki_push_api.py [hunk 3]
```diff
@@ -166,6 +166,31 @@ def __init__(self, *args):
   lib will automatically resolve the relation name inspecting the using the
   meta information of the charm
 
+Anytime the relation between a Loki provider charm and a Loki consumer charm is
+established a `LokiPushApiEndpointJoined` event is fired. In the consumer side
+is it possible to observe this event with:
+
```
Comment (sed-i) [inline lib/charms/loki_k8s/v0/loki_push_api.py:172]: I'm curious why should charm code care about these events. Maybe worth adding to docstring?

**Review by sed-i — APPROVED** [review]

File: lib/charms/loki_k8s/v0/loki_push_api.py (hunk 1 above)
Comment (Abuelodelanada) [reply lib/charms/loki_k8s/v0/loki_push_api.py:488]: Thanks, I forgot to deleted!

File: lib/charms/loki_k8s/v0/loki_push_api.py (hunk 2 above)
Comment (Abuelodelanada) [reply lib/charms/loki_k8s/v0/loki_push_api.py:494]: Thanks, I forgot to deleted!

File: lib/charms/loki_k8s/v0/loki_push_api.py (hunk 3 above)
Comment (Abuelodelanada) [reply lib/charms/loki_k8s/v0/loki_push_api.py:172]: We need to add or remove configuration in the consumer side (grafana-agent-operator).
Obviously docstrings can be improved!
Take a look at OP-369

**Pull request merged**